### PR TITLE
Add skeleton for Zuora (only store)

### PR DESCRIPTION
**Pull request by bruno**
Adds the skeleton for Zuora that implements tokenisation only and only for Australia for the time being. I am not 100% familiar with all the currencies and territories that are supported by Zuora, given their size, it's probably quite a lot.

### CODE REVIEW

**Comment by bruno** [comment]
Remote tests:

```
Finished in 4.306461 seconds.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
4 tests, 8 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
0.93 tests/s, 1.86 assertions/s
```

Unit tests:

```
Finished in 0.002512 seconds.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
3 tests, 9 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
1194.27 tests/s, 3582.80 assertions/s
```

File: lib/active_merchant/billing/gateways/zuora.rb [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+      def add_payment(post, payment, options = {})
+        if payment.respond_to?(:number)
+          post[:currency] = options[:currency]
+          post[:creditCard] = {
+            cardNumber: payment.number,
+            cardType: CARD_BRAND_MAP.fetch(payment.brand),
+            expirationMonth: format(payment.month, :two_digits),
+            expirationYear: format(payment.year, :four_digits),
+            securityCode: payment.verification_value
+          }.reject { |_, v| v.blank? }
```
Comment (hughevans) [inline lib/active_merchant/billing/gateways/zuora.rb]: Rails adds `#compact` to Hash if you wish to use it.

https://apidock.com/rails/Hash/compact

Comment (hughevans) [reply lib/active_merchant/billing/gateways/zuora.rb]: Except it doesn’t remove blank items. Carry on.

Comment (bruno) [reply lib/active_merchant/billing/gateways/zuora.rb]: But ActiveMerchant doesn't necessarily has access to all the nice railties.

**Pull request merged**